providers/implementations/keymgmt/ec_kmgmt.c

```diff
@@ -1006,10 +1006,10 @@ static void *ec_gen_init(void *provctx, int selection,
         gctx->libctx = libctx;
         gctx->selection = selection;
         gctx->ecdh_mode = 0;
-    }
-    if (!ec_gen_set_params(gctx, params)) {
-        OPENSSL_free(gctx);
-        gctx = NULL;
+        if (!ec_gen_set_params(gctx, params)) {
```

Comment:
Would it make sense to have `ec_gen_set_params()` check `gctx` for `NULL` (and then return an error) since `ec_gen_set_params()` is also callable via the `ossl_ec_keymgmt_functions[]` dispatch table?

Reply:
Well, I checked this table.  Some functions have a guard against a `NULL` argument, some do not.  Since this is part of the API, I am not sure where it is needed and where not.  Maybe someone from OTC can clarify this.  In my opinion, it does not make sense to call the set parameters callback with a `NULL` key.  If a defensive programming approach is desired, we should add checks to every function in this table (maybe also in the tables of other key management tables).  But of course this could lead to some performance regressions.

Reply:
Yup, I found the same, and I agree (we might want it on all or only some, and here and in other places) - ping @paulidale or @t8m

Reply:
I would keep the function as is for now. Certainly none of the functions in dispatch tables are supposed to be called with NULL context pointers and as you noticed we sometimes check and sometimes don't. Changing whether we do or don't for just a single function is IMO not going to improve the situation.